src/pet.js

```diff
@@ -0,0 +1,83 @@
+
+const vPet = {
+    minAge: 0,
+    maxAge: 30,
+    minHunger: 0,
+    maxHunger: 10,
+    minFitness: 0,
+    maxFitness: 10,
+    errorMessage: 'Your pet is no longer alive :('
+}
+
+class Pet {
+    constructor (name) {
+        this.name = name;
+        this.age = vPet.minAge;
+        this.hunger = vPet.minHunger;
+        this.fitness = vPet.maxFitness;
+        this.children = [];
+    }
+
+    get isAlive () {
+        return this.fitness > vPet.minFitness && this.hunger < vPet.maxHunger && this.age < vPet.maxAge;
+    }
+   
+    growUp () {
+        if (!this.isAlive) throw new Error (vPet.errorMessage);
+
+        this.age ++;
+        this.hunger += 5;
+        this.fitness -= 3;
+    }
+
+    walk () {
+        if (!this.isAlive) throw new Error (vPet.errorMessage);
+        
+        if ((this.fitness + 4) <= vPet.maxFitness) {
```

Comment:
Nice use of an if else statement here to make sure the pet never reaches over the maximum threshold for fitness.

Reply:
You don't need ton use the brackets here around this.fitness + 4 because the left and right hand side of the <= operator are resolved independently to the comparison itself. However, I think it makes it clearer to have them sometimes and saves you from having unexpected bugs in other areas of js where the order of operators is important.